frontend/src/components/graph/ArtifactNode.tsx

```diff
@@ -62,3 +62,27 @@ function ArtifactNode({ id, data }: ArtifactNodeProps) {
 }
 
 export default ArtifactNode;
+
+function getIcon(state: Artifact.State | undefined) {
+  if (state === undefined) {
+    return (
+      <div className='px-2 flex flex-col justify-center items-center rounded-l-lg'>
```

Comment:
All the class names are the same, maybe give them a name and reuse?

Reply:
Awesome point! Added a function for the wrapper div.